### DISCUSSION TITLE
¿Dónde queda el formulario inicial levantado por FAB button?

### DISCUSSION THREAD

**Discussion by seiturra** [Cotalker/documentation#173] [2. Questions and Answers] [answered]
Buenas a todos,

Voy a implementar un flujo que incluye FAB button para iniciar el flujo creando una task. Quisiera saber si el formulario levantado queda en alguna parte o será necesario hacer copia de este formulario "a mano" utilizando una etapa en la rutina que se ejecute. En caso de lo segundo, quisiera saber cómo se hace ya que me imagino que el contexto es distinto para este tipo de formularios.
Busqué [aquí](https://doc.cotalker.com/docs/documentation/admin/workflows/admin_workflow_groups/#channel-creation-section) pero no encontré esta información.

Saludos.

**Comment by rafaelwalter** [comment, marked as answer]
@SeIturra todos los formularios de inicio generan un canal automaticamente, y se envian en este canal. Este es el canal de la task que se crea con el formulario de inicio. Para encontrar el canal puedes mirar `task.channel` despues de creada.

**Reply by rafaelwalter** [reply]
@SeIturra esta es la documentacion sobre este comportamiento:

https://doc.cotalker.com/docs/documentation/admin/workflows/admin_workflow_required_survey#required-survey-for-a-new-task

**Reply by seiturra** [reply]
@rafaelwalter sólo una duda: mencionas que se genera un canal "y se envían en este canal". ¿Ese envío del formulario debe ser hecho a mano como lo hacemos actualmente con los bots de canal? ¿Se puede utilizar la etapa "Copiar mensajes" con el parámetro $VALUE#messages? Es un poco doloroso utilizar la etapa "Enviar Formulario" debido a que cada tipo de question se comporta distinto, por ejemplo, iterar si hay survey+survey o arrays y hay que aprender cómo son las entradas que esa etapa recibe para cada tipo de question. Si pudiésemos tener algo estándar para todos los tipos de questions sería mucho menos doloroso y fácil de aprender. 

Como sugerencia para estandarizar podríamos hacer que lo que se guarda en el campo process sea lo que la etapa Send Survey está preparada para recibir como input (actualmente, por ejemplo, para questions de propiedades, hay que traer la propiedad para pasar su name.code o para usuarios hay que buscarlos y pasar como parámetro su email). Otra alternativa sería que exista siempre el campo $VALUE#messages o algo similar para poder utilizar la etapa "Copiar mensajes" y copiar las respuestas de todo un formulario directamente.

---

Un feedback sobre ese punto de la documentación. Siguiendo los pasos, en el punto 6 no se entiende a qué "Name field" se refiere. Quizás en "follow the general procedure for making bots" debería haber un link a los pasos o mencionar que se debe crear una etapa "Crear Tarea" donde aparece el "Name field".

5. Next, select the Edit Routine button to create the necessary bot. Once inside the Routine builder dialog box, follow the general procedure for making bots keeping in mind the required configurations mentioned below.
6. In the Name field, press the add item button, and include something like the following:
$VALUE#answer|data|[find=>identifier=text_input_question_identifier]|process|0

**Reply by rafaelwalter** [reply]
@SeIturra 

- ¿Ese envío del formulario debe ser hecho a mano como lo hacemos actualmente con los bots de canal?
No. Es 100% automatica la creacion del canal y el envio del formulario en ese canal.

- ¿Se puede utilizar la etapa "Copiar mensajes" con el parámetro $VALUE#messages?
No es necesario.

**Reply by seiturra** [reply]
> Es 100% automatica la creacion del canal y el envio del formulario en ese canal.

Buenísimo! Gracias @rafaelwalter!

**Reply by rafaelwalter** [reply]
@Mauriceac en la documentaciond e start surveys deberiamos agregar antes del step 6 que se debe elegir el bot PBCreateTask, ya que el step 6 hace referencia a campos de ese bot.

**Reply by Mauriceac** [reply]
Corregido.
PR #190 

Gracias!
